### ISSUE TITLE
yo jhipster:upgrade doesn't work to upgrade from 4.0.0. to 4.0.1

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster#5116]
##### **Overview of the issue**

Running `yo jhipster:upgrade` doesn't work to upgrade from 4.0.0 to 4.0.1. 

It also still references `bower install`.

```
Successfully regenerated app with jhipster 4.0.1
Committed with message "Generated with JHipster 4.0.1"
Checked out branch "master"
Merging changes back to master...
Merge done!


I'm all done. Running npm install && bower install for you to install the required dependencies. If this fails, try running the command yourself.
```

##### **Reproduce the error**

```
git clone https://github.com/mraible/jhipster4-demo.git
cd jhipster4-demo
yo jhipster:upgrade
```

After running this, `git status` shows no changes. At the very least, I'd expect `package.json` to have an upgraded version.

##### **Browsers and Operating System**

java version "1.8.0_112"
Java(TM) SE Runtime Environment (build 1.8.0_112-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.112-b16, mixed mode)

git version 2.10.2

node: v6.9.5

npm: 3.10.10

yeoman: 1.8.5

yarn: 0.19.1

Docker version 1.13.0, build 49bf474

docker-compose version 1.10.0, build 4bd6f1a

**Comment by jdubois** [comment]
To be honnest I didn't test it...

**Comment by pascalgrimaud** [comment]
Same as Julien, I always prefer having full control on my git branch, so it's a sub generator I won't use

**Comment by jdubois** [comment]
Yes but we need to check if it still works, and correct it if necessary

**Comment by ctamisier** [comment]
I think I have the same problem from migrating from 3.12.1 to 4.0.1.
During the upgrade when the process should generate with version 4.0.1 but it still generates it with 3.12.1.
I was thinking my environment was wrong, so I tried to remove my generator and try to make everything clean but I wasn't able to upgrade with the upgrade generator.

```
Successfully regenerated app with jhipster 3.12.1
Committed with message "Generated with JHipster 3.12.1"
Checked out branch "develop"
Current code recorded as generated with version 3.12.1
Updating generator-jhipster. This might take some time...
Updated generator-jhipster to version 4.0.1
Checked out branch "jhipster_upgrade"
Cleaned up directory
Regenerating app with jhipster 4.0.1...

        ██  ██    ██  ████████  ███████    ██████  ████████  ████████  ███████
        ██  ██    ██     ██     ██    ██  ██          ██     ██        ██    ██
        ██  ████████     ██     ███████    █████      ██     ██████    ███████
  ██    ██  ██    ██     ██     ██             ██     ██     ██        ██   ██
   ██████   ██    ██  ████████  ██        ██████      ██     ████████  ██    ██

                            http://jhipster.github.io

Welcome to the JHipster Generator v3.12.1
Documentation for creating an application: https://jhipster.github.io/creating-an-app/
Application files will be generated in folder: /Users/tam/projects/mastermail
 ______________________________________________________________________________

  JHipster update available: 4.0.1 (current: 3.12.1)
  Run npm install -g generator-jhipster to update.
```

**Comment by jdubois** [comment]
In fact it works if you use the same technologies as JHipster 3.x 

- NPM for building
- Angular 1.x (with Gulp and Bower)

And it's broken with the new technologies, as nothing has been done to upgrade, biggest trouble being to use the Yarn commands instead of NPM

**Comment by jdubois** [comment]
In fact that's quite a lot of work to upgrade, so I'm marking a big warning on this sub-generator's documentation, and there will be an upgrade to be done here.

cc @lordlothar99 as you are the original author, do you have time to work on this?

**Comment by lordlothar99** [comment]
hi sorry guys for the delay, just went back from holidays ..
yeah sure I'll have a look !

**Comment by jdubois** [comment]
@lordlothar99 will you have time to work on this in the next days? I'm expecting a 4.0.4 release in one week, and I'd like to have this fixed, so if you don't have time I'll have a look

**Comment by lordlothar99** [comment]
I'll try to spend time on this today, but I can't find time I'll tell you tomorrow

**Comment by jdubois** [comment]
Great, thanks!

**Comment by lordlothar99** [comment]
Sorry guys, I'm not home yet... Seems like I won't have much time these days to work on this. Would you please get it done by yourself?

**Comment by xetys** [comment]
should we now remove tha warning in our documentation?

**Comment by pascalgrimaud** [comment]
@xetys : no, it's not working...

**Comment by xetys** [comment]
yeah....I learned this right now on my project....I just tried to update from 4.0.2 to 4.0.6....thanks git for rescuing my code :smile:

**Comment by jhhobbs** [comment]
I'm believe there may still be some issues.  When I try to upgrade, it appears the process never uses the newer version of JHipster, but uses the version the project is being updated from instead.

`$ yo jhipster:upgrade --target-version=4.3.0`

Output showing where process logs `Regenerating app with jhipster 4.3.0...` but then runs `Welcome to the JHipster Generator v4.0.3` generator:

...

Successfully regenerated app with jhipster 4.0.3
Committed with message "Generated with JHipster 4.0.3"
Checked out branch "master"
Current code recorded as generated with version 4.0.3
Updating generator-jhipster. This might take some time...
Updated generator-jhipster to version 4.3.0
Checked out branch "jhipster_upgrade"
Cleaned up directory
Regenerating app with jhipster 4.3.0...
 
        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝

                            http://jhipster.github.io

Welcome to the JHipster Generator v4.0.3
Documentation for creating an application: https://jhipster.github.io/creating-an-app/
Application files will be generated in folder: /development/projects/eclipse/workspace-neon/circulator
This is an existing project, using the configuration from your .yo-rc.json file 
to re-generate the project...

 ______________________________________________________________________________

  JHipster update available: 4.3.0 (current: 4.0.3)

**Comment by gmarziou** [comment]
@jhhobbs please open a new issue with all required details
